lib/usbd/usbd.c

```diff
@@ -224,6 +234,8 @@ bool usbd_is_vbus(usbd_device *dev){
 }
 
 void usbd_enable(usbd_device *dev, bool onoff){
+	if (!onoff)
+		usbd_disconnect(dev, false);
 	if (dev->backend->power_control)
 		dev->backend->power_control(dev, (onoff)? usbd_paActivate : usbd_paShutdown );
 }
```

Comment:
```
Application code
= usbd_disconnect(dev, true)
== dev->backend->disconnect(dev, true);
== usbd_enable(dev, false);
=== usbd_disconnect(dev, false);
==== usbd_enable(dev, true);
===== dev->backend->power_control(dev, usbd_paActivate);
==== dev->backend->disconnect(dev, false);
=== dev->backend->power_control(dev, usbd_paShutdown );
```

Look at the flow of control.
At the end, NOT disconnected with SHUTDOWN power control.